ffi/capi/src/units_converter.rs

```diff
@@ -0,0 +1,82 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_experimental::units::converter::UnitsConverter;
+use icu_experimental::units::converter_factory::ConverterFactory;
+use icu_experimental::units::measureunit::MeasureUnit;
+
+#[diplomat::bridge]
+pub mod ffi {
+    use crate::{errors::ffi::ICU4XError, provider::ffi::ICU4XDataProvider};
+    use alloc::boxed::Box;
+    use core::str;
+
+    #[diplomat::opaque]
+    /// An ICU4X Units Converter Factory object, capable of creating converters a [`ICU4XUnitsConverter`]
+    /// from [`ICU4XMeasureUnit`]s.
+    /// Also, it can parse the CLDR unit identifier (e.g. `meter-per-square-second`) and get the [`ICU4XMeasureUnit`].
+    #[diplomat::rust_link(icu::experimental::units::converter_factory::ConverterFactory, Struct)]
+    #[diplomat::rust_link(
+        icu::experimental::units::converter_factory::ConverterFactory,
+        Struct,
+        hidden
+    )]
+    pub struct ICU4XUnitsConverterFactory(pub ConverterFactory);
+
+    impl ICU4XUnitsConverterFactory {
+        /// Construct a new [`ICU4XUnitsConverterFactory`] instance.
+        #[diplomat::rust_link(
+            icu::experimental::units::converter_factory::ConverterFactory::new,
+            FnInStruct
+        )]
+        #[diplomat::attr(all(supports = constructors, supports = fallible_constructors), constructor)]
+        pub fn create(
+            provider: &ICU4XDataProvider,
+        ) -> Result<Box<ICU4XUnitsConverterFactory>, ICU4XError> {
+            Ok(Box::new(ICU4XUnitsConverterFactory(call_constructor!(
+                ConverterFactory::new [r => Ok(r)],
+                ConverterFactory::try_new_with_any_provider,
+                ConverterFactory::try_new_with_buffer_provider,
+                provider,
+            )?)))
+        }
+
+        /// Creates a new [`ICU4XUnitsConverter`] from the input and output `ICU4XMeasureUnit`s.
+        /// Returns `None` if the conversion between the two units is not possible.
```

Comment:
to be language-agnostic:
```suggestion
        /// Returns nothing if the conversion between the two units is not possible.
```

Reply:
done